instrumentation/opentelemetry_req/test/opentelemetry_req_test.exs

```diff
@@ -1,4 +1,120 @@
 defmodule OpentelemetryReqTest do
-  use ExUnit.Case
+  use ExUnit.Case, async: true
+
   doctest OpentelemetryReq
+
+  require Record
+
+  for {name, spec} <- Record.extract_all(from_lib: "opentelemetry/include/otel_span.hrl") do
+    Record.defrecord(name, spec)
+  end
+
+  for {name, spec} <- Record.extract_all(from_lib: "opentelemetry_api/include/opentelemetry.hrl") do
+    Record.defrecord(name, spec)
+  end
+
+  setup do
+    :otel_simple_processor.set_exporter(:otel_exporter_pid, self())
+
+    :ok
```

Comment:
you may also consider returning:

```elixir
    pid = self()

    adapter = fn req ->
      send(pid, {:req, req})
      {req, Req.Response.new(status: 204)}
    end

    req =
      Req.new(adapter: adapter)
      |> OpentelemetryReq.attach()

    %{req: req}
```

which would remove some duplication from the tests. But then again such duplication doesn't hurt too much either. :)